pkgs/build-support/java/canonicalize-jar.sh

```diff
@@ -0,0 +1,28 @@
+# Canonicalize the manifest & repack with deterministic timestamps.
+canonicalizeJar() {
+    local input="$(realpath -sm -- "$1")"
+    local outer="$(pwd)"
+    # -qq: even quieter
+    @unzip@ -qq "$input" -d "$input-tmp"
+    canonicalizeJarManifest "$input-tmp/META-INF/MANIFEST.MF"
+    # Sets all timestamps to Jan 1 1980, the earliest mtime zips support.
+    find -- "$input-tmp" -exec touch -t 198001010000.00 {} +
+    rm "$input"
+    pushd "$input-tmp" 2>/dev/null
+    # -q|--quiet, -r|--recurse-paths
+    # -o|--latest-time: canonicalizes overall archive mtime
+    # -X|--no-extra: don't store platform-specific extra file attribute fields
+    @zip@ -qroX "$outer/tmp-out.jar" . 2> /dev/null
```

Comment:
Could use the `-m` option and have
```sh
@zip@ -qroXm "$outer/tmp-out.jar" "$input" 2> /dev/null
```
perhaps? Then could remove the `mv "$outer/tmp-out.jar" "$input"` below and change `rm -rf "$input-tmp"` to `rmdir "$input-tmp"`.

Reply:
This won't work because then you'll record the input directory name. We need to be inside `$input-tmp".